Xwt.WPF/Xwt.WPFBackend/AccessibleBackend.cs

```diff
@@ -46,31 +53,87 @@ public void EnableEvent (object eventId)
 
 		public void Initialize (IWidgetBackend parentWidget, IAccessibleEventSink eventSink)
 		{
-			var parent = parentWidget as WidgetBackend;
-			Initialize (parent.Widget, eventSink);
+			Initialize (parentWidget.NativeWidget, eventSink);
 		}
 
 		public void Initialize (object parentWidget, IAccessibleEventSink eventSink)
 		{
 			this.element = parentWidget as UIElement;
 			if (element == null)
 				throw new ArgumentException ("Widget is not a UIElement");
+			this.eventSink = eventSink;
 		}
 
 		public void InitializeBackend (object frontend, ApplicationContext context)
 		{
+			this.context = context;
+		}
+
+		internal void PerformInvoke ()
+		{
+			context.InvokeUserCode (() => eventSink.OnPress ());
 		}
 
+		// The following child methods are only supported for Canvas based widgets
 		public void AddChild (object nativeChild)
 		{
+			var peer = nativeChild as AutomationPeer;
+			var canvas = element as CustomCanvas;
+			if (peer != null && canvas != null)
+				canvas.AutomationPeer?.AddChild (peer);
 		}
 
 		public void RemoveAllChildren ()
 		{
+			var canvas = element as CustomCanvas;
+			if (canvas != null)
+				canvas.AutomationPeer?.RemoveAllChildren ();
 		}
 
 		public void RemoveChild (object nativeChild)
 		{
+			var peer = nativeChild as AutomationPeer;
+			var canvas = element as CustomCanvas;
+			if (peer != null && canvas != null)
+				canvas.AutomationPeer?.RemoveChild (peer);
+		}
+
+		public static AutomationControlType RoleToControlType (Role role)
+		{
+			switch (role) {
+			case Role.Button:
+			case Role.MenuButton:
+			case Role.ToggleButton:
+				return AutomationControlType.Button;
+			case Role.CheckBox:
+				return AutomationControlType.CheckBox;
+			case Role.RadioButton:
+			case Role.RadioGroup:
+				return AutomationControlType.RadioButton;
+			case Role.ComboBox:
+				return AutomationControlType.ComboBox;
+			case Role.List:
+				return AutomationControlType.List;
+			case Role.Popup:
+			case Role.ToolTip:
+				return AutomationControlType.ToolTip;
+			case Role.ToolBar:
+				return AutomationControlType.ToolBar;
+			case Role.Label:
+				return AutomationControlType.Text;
+			case Role.Link:
+				return AutomationControlType.Hyperlink;
+			case Role.Image:
+				return AutomationControlType.Image;
+			case Role.Cell:
+				return AutomationControlType.DataItem;
+			case Role.Table:
+				return AutomationControlType.Table;
```

Comment:
`AutomationControlType.Table` indicates that the table contains only text, I think this should be [`AutomationControlType.DataGrid`](https://docs.microsoft.com/de-de/dotnet/api/system.windows.automation.controltype.datagrid?view=netframework-4.7.1#remarks) because we support any cell type.